doc/code/qml_workflow.rst

```diff
@@ -0,0 +1,8 @@
+qml.workflow
```

Comment:
I would avoid creating a module without a plan shared and accepted by the team and the product team. Without it I am afraid that we will have to deprecate it in some months after we realize that it is not useful or does not fit our needs. In the meantime, I would add that in the interface module where `execution.py` is.

Reply:
I agree that this `workflow` module is coming out of nowhere, but I don't feel like `interfaces` is also not the best home (since we are trying to isolate new code from old code). While I am not certain that this is where it will end up, this certainly enables us developers now while maintaining the separation needed to get this into the desired state once more planning/discussion has happened.

Perhaps we should remove the documentation until we have more certainty about how everything will end up? `ExecutionConfig` isn't imported at the top level as of this PR.

Reply:
Simple solution for now: Just don't build it on the website.

That will keep this an internal implementation detail until the module is better fleshed out. Source code locations are purely implementation details; import paths for user-facing objects are product details.

As this object's only "interface" connection is a single property called "framework" and it will interact with qnodes, quantum scripts, and devices, it does not belong in `interfaces` at all.

Reply:
I don't think it is an implementation detail when we create a new module even if we do not add documentation. We already have two execute pipelines, I am not in favour to add a third partially working one. This project (the module) is a great occasion to clean and create a better pipeline and we should plan accordingly and discuss the main points. I don't think the planning can be done until we have merged a working device. It should be postpone until:

1. The new device work is in a good state and unit tests are added (avoid relying on integration tests for coverage).
2. The workflow module and a new pipeline are approved by @josh146 and @trbromley 

Or all the workflow PRs are made on a separate branch `worflow` that is not merged in master until we have a satisfying new execution pipeline.

Reply:
fair enough. if we intend on _not_ having something complete/functional where we maintain a WIP pipeline, I'd prefer to do this work on a `workflow` branch